src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,200 @@
+            throw new DataProcessingException("Can't get list of cars from DB", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car of " + car, e);
+        }
+        deleteDriversFromCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE "
+                + "WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id, model, manufacturer_id, name, country, cd.driver_id "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd "
+                + "ON c.id = cd.car_id "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE "
+                + "AND m.is_deleted = FALSE";
+        List<Car> carsByDriverList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carsByDriverList.add(getCar(resultSet));
+            }
+            return carsByDriverList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get list of cars by driver id "
+                    + driverId, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        return new Car(resultSet.getObject("id", Long.class),
+                resultSet.getString("model"),
+                new Manufacturer(resultSet.getObject("manufacturer_id", Long.class),
+                        resultSet.getString("name"),
+                        resultSet.getString("country")),
+                getDrivers(resultSet.getObject("id", Long.class)));
```

Comment:
![image](https://user-images.githubusercontent.com/117590814/214822269-66430563-2757-4c9f-9880-23caf9fc2d80.png)
just remove getDrivers call from this method and call separately in the end of dao methods